### ISSUE TITLE
 Failed to open given input: Function not implemented pl.droidsonroids.gif.GifInfoHandle.extractNativeFileDescriptor

### ISSUE THREAD

**Issue by antocara** [koral--/android-gif-drawable#701]
Hi, recently using this library, gave me the following error. Below I attach all the data that Firebase throws about the error. Thanks for the great job.
```
Caused by pl.droidsonroids.gif.GifIOException: GifError 101: Failed to open given input: Function not implemented
       at pl.droidsonroids.gif.GifInfoHandle.extractNativeFileDescriptor(GifInfoHandle.java)
       at pl.droidsonroids.gif.GifInfoHandle.openFileDescriptor(GifInfoHandle.java:83)
       at pl.droidsonroids.gif.GifInfoHandle.<init>(GifInfoHandle.java:64)
       at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:159)
       at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:94)
       at es.cashconverters.compraventa.presentation.views.wizard.WizardStepFragment.onActivityCreated(WizardStepFragment.kt:29)
       at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2460)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1483)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
       at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:802)
       at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManager.java:2625)
       at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
       at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2366)
       at androidx.fragment.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:2243)
       at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:654)
       at androidx.fragment.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:146)
       at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
       at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
       at androidx.viewpager.widget.ViewPager.onMeasure(ViewPager.java:1622)
       at android.view.View.measure(View.java:18794)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5951)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
       at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5951)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1465)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:748)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:630)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5951)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5951)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1465)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:748)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:630)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5951)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
       at com.android.internal.policy.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2643)
       at android.view.View.measure(View.java:18794)
       at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2100)
       at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1216)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1452)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
       at android.view.Choreographer.doCallbacks(Choreographer.java:670)
       at android.view.Choreographer.doFrame(Choreographer.java:606)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5421)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

The version of the library is: **pl.droidsonroids.gif:android-gif-drawable:1.2.17**
and de device is:  **Lg Nexus 5X with Android 6.0.1**
The app is **developed in Kotlin**.
Regards, thanks

**Comment by koral--** [comment]
Could you attach code at es.cashconverters.compraventa.presentation.views.wizard.WizardStepFragment.onActivityCreated(WizardStepFragment.kt:29) ?

**Comment by abahl817** [comment]
Hey @koral-- , I had a similar issue and I'm also attaching the code I used at the desired line. Hope this helps.
`GifDrawable gifDrawable = new GifDrawable(getResources(), R.mipmap.edurev_gif);
            gifImageView.setBackground(gifDrawable);
            gifDrawable.setLoopCount(1);
            gifDrawable.start();
            gifDrawable.addAnimationListener(new AnimationListener() {
                @Override
                public void onAnimationCompleted(int loopNumber) {
                    checkForAppUpdates();
                }
            });`

**Comment by koral--** [comment]
OK, I'll check that.

**Comment by antocara** [comment]
> Could you attach code at es.cashconverters.compraventa.presentation.views.wizard.WizardStepFragment.onActivityCreated(WizardStepFragment.kt:29) ?

Basically the code inside the WizardStepFragment is:

```
override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         label_header_wizard_step.text = context?.getString(wizardStep.headerStringId)    
         val gifFromResource = GifDrawable(resources, wizardStep.gifId)
         image_wizard_step.setImageDrawable(gifFromResource)
     }
```

and the xml with de ImageView for display gif is:
```
<pl.droidsonroids.gif.GifImageView
            android:scaleType="centerCrop"
            android:id="@+id/image_wizard_step"
            android:layout_marginEnd="10dp"         />
```
Thanks

**Comment by koral--** [comment]
I could not repro this issue, but potential fix is included in version 1.2.20-SNAPSHOT.

**Comment by koral--** [comment]
Possible fix in https://github.com/koral--/android-gif-drawable/commit/5c69d15b88fdefd15cd2bde15bbfc219a38e716e
